File: Examples.VB/Features/Charts/DataLabel/ConfigDataLabelAngle.vb
Namespace Features.Charts.DataLabel
    Public Class ConfigDataLabelAngle
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim worksheet As IWorksheet = workbook.Worksheets(0)

            Dim shape As IShape = worksheet.Shapes.AddChart(ChartType.ColumnClustered, 250, 20, 360, 230)
            worksheet.Range("A1:B5").Value = New Object(,) {
                {Nothing, "S1"},
                {"Item1", -20},
                {"Item2", 30},
                {"Item3", 50},
                {"Item3", 40}
            }
            shape.Chart.SeriesCollection.Add(worksheet.Range("A1:B5"), RowCol.Columns, True, True)

            Dim series1 As ISeries = shape.Chart.SeriesCollection(0)
            series1.HasDataLabels = True

            ' set series1's all data labels' angle.
            series1.DataLabels.Orientation = 45

            ' set series1's specific data label's angle.
            series1.DataLabels(2).Orientation = -45
        End Sub
    End Class
End Namespace

File: Examples/Features/Charts/DataLabel/ConfigDataLabelAngle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Charts.DataLabel
{
    public class ConfigDataLabelAngle : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            IWorksheet worksheet = workbook.Worksheets[0];

            GrapeCity.Documents.Excel.Drawing.IShape shape = worksheet.Shapes.AddChart(GrapeCity.Documents.Excel.Drawing.ChartType.ColumnClustered, 250, 20, 360, 230);
            worksheet.Range["A1:B5"].Value = new object[,]
            {
                {null, "S1"},
                {"Item1", -20},
                {"Item2", 30},
                {"Item3", 50 },
                {"Item3", 40 }
            };
            shape.Chart.SeriesCollection.Add(worksheet.Range["A1:B5"], GrapeCity.Documents.Excel.Drawing.RowCol.Columns, true, true);
            GrapeCity.Documents.Excel.Drawing.ISeries series1 = shape.Chart.SeriesCollection[0];
            series1.HasDataLabels = true;

            //set series1's all data labels' angle.
            series1.DataLabels.Orientation = 45;

            //set series1's specific data label's angle.
            series1.DataLabels[2].Orientation = -45;

        }

        public override bool IsNew
        {
            get
            {
                return true;
            }
        }

        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }

        public override bool ShowScreenshot
        {
            get
            {
                return true;
            }
        }
    }
}
